core/src/ledger/ibc/mod.rs

```diff
@@ -255,3 +260,37 @@ pub struct ValidationParams {
     /// Upgrade path
     pub upgrade_path: Vec<String>,
 }
+
+/// Get the IbcToken from the source/destination ports and channels
+pub fn received_ibc_token(
+    token: &Address,
+    trace_path: Option<String>,
+    src_port_id: &PortId,
+    src_channel_id: &ChannelId,
+    dest_port_id: &PortId,
+    dest_channel_id: &ChannelId,
+) -> Result<Address, Error> {
+    if let Some(trace_path) = trace_path {
+        let mut ibc_denom =
+            PrefixedDenom::from_str(&format!("{}/{}", trace_path, token))
+                .map_err(|e| {
+                    Error::Denom(format!("Trace path is invalid: error {e}"))
+                })?;
+        if is_receiver_chain_source(
+            src_port_id.clone(),
+            src_channel_id.clone(),
+            &ibc_denom,
+        ) {
+            let prefix =
+                TracePrefix::new(src_port_id.clone(), src_channel_id.clone());
+            ibc_denom.remove_trace_prefix(&prefix);
```

Comment:
ah so this is how we "pop off" traces from the path, to eventually redeem native tokens!